### PR TITLE
Registering optional fields and tags with tchannel server endpoint

### PR DESCRIPTION
**Pull request by ravirajj**

### CODE REVIEW

File: runtime/tchannel_server.go [hunk 1]
```diff
@@ -98,34 +98,42 @@ func NewTChannelRouter(registrar tchannel.Registrar, g *Gateway) *TChannelRouter
 func (s *TChannelRouter) Register(
 	endpointID, handlerID, method string,
 	h TChannelHandler,
+	fields []zapcore.Field,
```
Comment (ChuntaoLu) [inline runtime/tchannel_server.go]: This method is getting completely out of hand, the `register` semantic is simple, and yet now it is bloated with 6 arguments and returns an endpoint? It does way more than what its name means.

Comment (Raynos) [reply runtime/tchannel_server.go]: Taking an options struct is fine. 

You could also pass in the `Endpoint` similar to the `Router` API for http.

Comment (ravirajj) [reply runtime/tchannel_server.go]: Passing a `TChannelEndpoint` to `Register` like `Router` for http.

**Comment by ravirajj** [comment]
Abandoning in favor of #237.

**Pull request closed without merging**